apis/core/v1beta1/openstackcontrolplane_types.go

```diff
@@ -388,6 +388,23 @@ type NovaSection struct {
 	//+operator-sdk:csv:customresourcedefinitions:type=spec
 	// Template - Overrides to use when creating the Nova services
 	Template novav1.NovaSpec `json:"template,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// APIOverride, provides the ability to override the generated manifest of several child resources.
+	APIOverride Override `json:"apiOverride,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// CellOverride, provides the ability to override the generated manifest of several child resources
+	// for a nova cell.
+	CellOverride map[string]NovaCellOverrideSpec `json:"cellOverride,omitempty"`
```

Comment:
> ```yaml
>     nova:
>       enabled: true
>       cellOverride:
>         cell1:
>           noVNCProxy:
>             route:
>               spec:
>                 tls
>       template:
>         cellTemplates:
>           cell0:
>           cell1:
>             noVNCProxyServiceTemplate:
>               replicas: 1
> ```
> 
> Form API UX perspective this feels odd as we now have two parallel structures to define cells. The end user need to mentally merge the two parallel structures by key names to get the full picture of the configuration of a cell.
> 
> From implementation perspective openstack-operator needs to learn special rules about nova cells. E.g. that cell0 never have compute nodes and therefore never have running VMs and therefore it does not make sense to have noVNCProxy deployed. Today nova-operator prevents cell0 to have noVNCProxy service deployed. https://github.com/openstack-k8s-operators/nova-operator/blob/8cf10addb78028c2387e0208aeaa9b72d7920226/api/v1beta1/novanovncproxy_webhook.go#L103-L115 So far this information and the related validation is encapsulated in the nova-operator but the parallel cell route configuration here now requires that the openstack-operator also ensure no route is created for noVNCProxy in cell0.

not sure I get that, there is no knowledge required in the openstack-operator on cell specifics. The nova-operator won't create a novnxproxy for cell0 and therefore there is no service for a novncproxy. The openstack-operator therefore won't create a route, or am I missing something here?

Reply:
Ahh you are right. No k8s service means no route creation. So there won't be an unused Route object for cell0. So the following config is always is ignored as cell0 has no NoVNCProxy.
``` yaml
    nova:
      enabled: true
      cellOverride:
        cell0:
          noVNCProxy:
            route:
              spec:
                tls
```
To avoid confusion from the user we at least needs to document it that adding route override for cell0 is ignored. Or we can add a validation webhook check to reject such config. In both case we duplicating pieces of nova-operator. 
[1] https://github.com/openstack-k8s-operators/nova-operator/blob/8cf10addb78028c2387e0208aeaa9b72d7920226/api/v1beta1/novacell_types.go#L75-L80
[2] https://github.com/openstack-k8s-operators/nova-operator/blob/8cf10addb78028c2387e0208aeaa9b72d7920226/api/v1beta1/novanovncproxy_webhook.go#L105-L114

Reply:
I have updated the comment in the CRD